Source/WebCore/fileapi/ThreadableBlobRegistry.cpp

```diff
@@ -175,6 +175,19 @@ void ThreadableBlobRegistry::registerInternalBlobURLForSlice(const URL& newURL,
     });
 }
 
+String ThreadableBlobRegistry::blobType(const URL& url)
+{
+    if (isMainThread())
+        return blobRegistry().blobType(url);
+
+    String result;
+    callOnMainThreadAndWait([url = url.isolatedCopy(), &result] {
+        result = blobRegistry().blobType(url);
```

Comment:
We probably want to isolate copy result, this should hopefully not create any memory allocation.
Or simply return early if not main thread.